picard/ui/widgets/scripttextedit.py

```diff
@@ -145,50 +186,87 @@ def enable_completer(self):
         self.popup_shown = False
 
     def insert_completion(self, completion):
+        if not completion:
+            return
         tc = self.cursor_select_word()
+        if completion.startswith('$'):
+            completion += '('
         tc.insertText(completion)
+        pos = tc.position()
+        # Peek at the next character to include it in the replacement
+        if not tc.atEnd():
+            tc = self.textCursor()
+            tc.setPosition(pos + 1, QTextCursor.KeepAnchor)
+            first_char = completion[0]
+            next_char = tc.selectedText()
+            if (first_char == '$' and next_char == '(') or (first_char == '%' and next_char == '%'):
+                tc.removeSelectedText()
+            else:
+                tc.setPosition(pos)  # Reset position
         self.setTextCursor(tc)
         self.popup_hide()
 
     def popup_hide(self):
         self.completer.popup().hide()
 
-    def cursor_select_word(self):
+    def cursor_select_word(self, full_word=True):
         tc = self.textCursor()
         current_position = tc.position()
         tc.select(QTextCursor.WordUnderCursor)
+        if not full_word:
+            tc.setPosition(current_position, QTextCursor.KeepAnchor)
         selected_text = tc.selectedText()
+        # Check for start of function or end of variable
+        if selected_text and selected_text[0] in ('(', '%'):
+            current_position -= 1
```

Comment:
May be check `current_position` isn't 0 before? Useful?

Reply:
I think it is at least better to check this. I could not reproduce a case for this, but we would rely on `tc.selectedText()` never returning something at position 0 which matches the condition. So better to check for > 0